### PR TITLE
PAT_437 Nominal search (probation).

### PR DESCRIPTION
**Pull request by lbennett-moj**
Search form and validation and offender search client
Integration and unit tests
todo: docker-compose, results processing and display

### CODE REVIEW

File: server/utils/nunjucksSetup.js [hunk 1]
```diff
@@ -17,6 +17,18 @@ module.exports = (app, path) => {
     return null
   })
 
+  njkEnv.addFilter('findErrorMap', (errorKeys, formFieldId) => {
```
Comment (lbennett-moj) [inline server/utils/nunjucksSetup.js]: allow user of error message map with nunjucks  components

File: server/data/elite2ClientBuilder.test.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const elite2ClientBuilder = require('../../server/data/elite2ClientBuilder')
 
 const context = { username: 'test', token: 'token-1' }
 
-jest.mock('cls-hooked')
+// jest.mock('cls-hooked')
```
Comment (lbennett-moj) [inline server/data/elite2ClientBuilder.test.js]: was just checking this was required - I'll remove on next commit

File: assets/sass/local.sass [hunk 3]
```diff
@@ -254,3 +254,14 @@ p.line-headings
   background: govuk-tint(#5694CA, 90)
   width: 577px
   height: 200px
+
+.idInputWidth
```
Comment (yasinmustafa) [inline assets/sass/local.sass]: Can we use hyphen delimited for css class naming? e.g. .id-input-width

Comment (lbennett-moj) [reply assets/sass/local.sass]: no problem - I'll update that in the next commit for this story

File: server/routes/nominalSearch.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    !isBlank(searchForm.pnc)
+
+  if (!isBlank(searchForm.day) || !isBlank(searchForm.month) || !isBlank(searchForm.year)) {
+    searchForm.dob = `${searchForm.year}-${searchForm.month}-${searchForm.day}`
+  }
+  return searchForm
+}
+
+// eslint-disable-next-line no-unused-vars
+module.exports = ({ offenderService, pathfinderService }) => router => {
```
Comment (yasinmustafa) [inline server/routes/nominalSearch.js]: I guess were passing the pathfinderService in for future use? minor so happy to approve

File: server/data/elite2ClientBuilder.test.js (hunk 2 above)
Comment (yasinmustafa) [reply server/data/elite2ClientBuilder.test.js]: OK

File: server/routes/nominalSearch.js (hunk 4 above)
Comment (lbennett-moj) [reply server/routes/nominalSearch.js]: yes - will be needing it for the next part

**Pull request merged**